src/Workspaces/Core/Portable/Rename/RenameUtilities.cs

```diff
@@ -108,8 +109,24 @@ internal static IEnumerable<Document> GetDocumentsAffectedByRename(ISymbol symbo
         /// </summary>
         private static bool ShouldRenameOnlyAffectDeclaringProject(ISymbol symbol)
         {
-            // Explicit interface implementations can cascade to other projects
-            return symbol.DeclaredAccessibility == Accessibility.Private && !symbol.ExplicitInterfaceImplementations().Any();
+            if (symbol.DeclaredAccessibility == Accessibility.Private)
+            {
+                // Explicit interface implementations can cascade to other projects
+                // we check if the symbol is an override. If it's overriding something (even if private) it's an error, but we can rename across that error.
+                // https://github.com/dotnet/roslyn/issues/25682
+                if (symbol.ExplicitInterfaceImplementations().Any() || symbol.IsOverride)
+                {
+                    return false;
+                }
+                else
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                return false;
+            }
```

Comment:
consider writing as:

```c#
            if (symbol.DeclaredAccessibility != Accessibility.Private)
            {
                // non-private members can influence other projects.
                return false;
            }

            if (symbol.ExplicitInterfaceImplementations().Any())
            {
                // Explicit interface implementations can cascade to other projects
                return false;
            }

            if (symbol.IsOverride)
            {
                 // private-overrides aren't actually legal.  But if we see one, we tolerate it and search other projects in case
                 // they override it.
                 return false;
            }

            return true;
        }
```